src/shells.ts

```diff
@@ -1,30 +1,39 @@
 import * as vscode from 'vscode';
 
-export enum OS {
-  WINDOWS = 1,
-  LINUX = 2
-}
+// eslint-disable-next-line @typescript-eslint/naming-convention
+type systems = 'windows' | 'linux' | 'osx' | 'linuxbase';
 
-const windowsDefaultShell = 'powershell.exe';
-const linuxDefaultShell = '/bin/bash';
+// Default shells in vscode
+const defaultShells = {
+  windows: 'powershell.exe',
+  linux: '$SHELL',
+  osx: '$SHELL'
+};
 
+function getShell(os: Exclude<systems, 'linuxbase'>) {
 
-export function getShell(os: OS) {
   const integratedShell = vscode.workspace.getConfiguration('terminal.integrated.shell');
 
-  if (os === OS.WINDOWS) {
-    const windowsShell = integratedShell?.get<string>('windows');
-    if (windowsShell) {
-      return windowsShell;
-    }
-    return windowsDefaultShell;
+  const shell = integratedShell?.get<string>(os);
+  if (shell) {
+    return shell;
+  }
+
+  return defaultShells[os];
+}
+
+export function shellScript(os: Exclude<systems, 'osx'>) {
+  if (os === 'windows') {
+    return `${getShell('windows')}`;
+  }
```

Comment:
Since we don't know whether the local shell setting is valid on the remote, we should fallback nicely there. For example I may have git bash installed locally and prefer that, but the remote could not have it installed.

I'm not the greatest at powershell but I'm envisioning something like this:

```ts
if (os === 'windows') {
  const shell = getShell('windows');
  if (shell) {
    return `runIfExists(${shell}) || ${defaultShells['windows']}`;
  }
  return `defaultShells['windows']`;
}
```

Reply:
i think i know what you mean here. The way we are calling the shell actually does that. If a executable don't exist it fallsback to powershell (with the -NoExit argument). Even if the shell where we are executing the command is not powershell.

```
powershell -NoExit -Command cd dev \\; shell_windows
```